### PR TITLE
feat(StatusModal): Add edit avatar button

### PR DESCRIPTION
**Pull request by borismelnik**
Part of https://github.com/status-im/status-desktop/issues/3734

### CODE REVIEW

File: src/StatusQ/Popups/statusModal/StatusModalHeader.qml [hunk 1]
```diff
@@ -36,6 +38,18 @@ Rectangle {
...
+        anchors.bottom: parent.bottom
+        width: parent.width
+        height: parent.radius
+        color: parent.color
+
+        StatusModalDivider {
+            anchors.bottom: parent.bottom
+            width: parent.width
+        }
+    }
```
Comment (PascalPrecht) [inline src/StatusQ/Popups/statusModal/StatusModalHeader.qml:51]: Any reason this was moved?

File: src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml [hunk 2]
```diff
@@ -92,11 +117,12 @@ Row {
...
                 elide: statusImageWithTitle.titleElide
                 color: Theme.palette.directColor1
-                width: !editButton.visible ? parent.width : 
-                      parent.width - editButton.width - parent.spacing
+                width: !editButton.visible ? parent.width :
+                                             parent.width - editButton.width - parent.spacing
             }
             StatusFlatRoundButton {
                 id: editButton
+                objectName: "editAvatarbButton"
```
Comment (PascalPrecht) [inline src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:125]: Is this needed?

Comment (PascalPrecht) [reply src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:125]: If yes, what is it for?

File: src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml [hunk 3]
```diff
@@ -64,12 +66,35 @@ Row {
...
+
+            StatusRoundButton {
+                type: StatusRoundButton.Type.Secondary
+
+                anchors.right: parent.right
+                anchors.bottom: parent.bottom
+                anchors.bottomMargin: -3
+
+                width: 15
+                height: 15
```
Comment (PascalPrecht) [inline src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:84]: I believe this has to be bigger though. Looking at the figma, the button is 16x16 and also, you can see in the UI that it's super small:
Screenshot 2021-10-13 at 14 03 52

Comment (PascalPrecht) [reply src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:84]: Keep in mind that the `border.width` wont affect the overall size

Comment (PascalPrecht) [reply src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:84]: Also, it looks like it isn't positioned correctly. It's hanging "off" a but too much

File: src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml [hunk 4]
```diff
@@ -64,12 +66,35 @@ Row {
...
+
+                border.color: Theme.palette.indirectColor1
```
Comment (PascalPrecht) [inline src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:86]: I don't think we can do that and have to do no border here. It's hard to provide a border color with alpha channel that works across all backgrounds (avatar images):
Screenshot 2021-10-13 at 14 04 00

So we either have to do a hard check against `Theme.palette.name` and give it a solid border color based on theme, or we remove it entirely

File: src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml [hunk 5]
```diff
@@ -64,12 +66,35 @@ Row {
...
+                border.width: 1
+
+                icon.name: "tiny/edit"
+                icon.width: 8
+                icon.height: 8
```
Comment (PascalPrecht) [inline src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:91]: Visually, this seems to be too tiny. It's possible that there's a padding white space in the icon source, so wie might need to bump up the width and height here a little.

File: src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml [hunk 6]
```diff
@@ -12,8 +12,10 @@ Row {
     property int titleElide: Text.ElideRight
     property int subTitleElide: Text.ElideRight
     property bool editable: false
+    property bool editAvatar: false
```
Comment (PascalPrecht) [inline src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:15]: Another nitpick: how about naming this: `headerImageEditable` instead across all boards?

Comment (borismelnik) [reply src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:15]: no problem :)

File: src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml (hunk 4 above)
Comment (borismelnik) [reply src/StatusQ/Popups/statusModal/StatusImageWithTitle.qml:88]: okay

File: src/StatusQ/Popups/statusModal/StatusModalHeader.qml (hunk 1 above)
Comment (borismelnik) [reply src/StatusQ/Popups/statusModal/StatusModalHeader.qml:51]: bc of `z` order this line hide part of header.

**Pull request merged**